docs/source/user_guide/timeseries.ipynb

```diff
@@ -0,0 +1,374 @@
+    "        y=y_train,\n",
+    "        mode=\"lines+markers\",\n",
+    "        name=\"Temperature (C)\",\n",
+    "        line=dict(color=\"#1f77b4\"),\n",
+    "    )\n",
+    "]\n",
+    "# Let plotly pick the best date format.\n",
+    "layout = go.Layout(\n",
+    "    title={\"text\": \"Min Daily Temperature, Melbourne 1980-1989\"},\n",
+    "    xaxis={\"title\": \"Time\"},\n",
+    "    yaxis={\"title\": \"Temperature (C)\"},\n",
+    ")\n",
+    "\n",
+    "go.Figure(data=data, layout=layout)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Running AutoMLSearch\n",
+    "\n",
+    "`AutoMLSearch` for time series problems works very similarly to the other problem types with the exception that users need to pass in a new parameter called `problem_configuration`.\n",
+    "\n",
+    "The `problem_configuration` is a dictionary specifying the following values:\n",
+    "\n",
+    "* **forecast_horizon**: The number of time periods we are trying to forecast. In this example, we're interested in predicting weather for the next 7 days, so the value is 7.\n",
+    "\n",
+    "* **gap**: The number of time periods between the end of the training set and the start of the test set. For example, in our case we are interested in predicting the weather for the next 7 days with the data as it is \"today\", so the gap is 0. However, if we had to predict the weather for next week with the data as it was on the previous Friday, the gap would be 2. It is important to select a value that matches the realistic delay between the forecast date and the most recently avaliable data that can be used to make that forecast.\n",
+    "\n",
+    "* **max_delay**: The maximum number of rows to look in the past from the current row in order to compute features. In our example, we'll say we can use the previous week's weather to predict the current week's.\n",
+    "\n",
+    "* **date_index**: The column of the training dataset that contains the date corresponding to each observation. Currently, this parameter is only used by some time-series specific models so in this example, we are passing in `None`.\n",
+    "\n",
+    "### Visualization of forecast horizon and gap\n",
+    "![forecast and gap](ts_viz/ts_parameter_viz.png)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from evalml.automl import AutoMLSearch\n",
+    "\n",
+    "automl = AutoMLSearch(X_train, y_train, problem_type=\"time series regression\",\n",
+    "                      max_batches=1,\n",
+    "                      problem_configuration={\"gap\": 0, \"max_delay\": 7, \"forecast_horizon\": 7,\n",
+    "                                             \"date_index\": None},\n",
+    "                      )"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "automl.search()"
```

Comment:
When this is run in the docs, the XGBoost regressor fails with the error `ARIMA regressor requires input data X to have a datetime column specified by the 'date_index' parameter. If not it will look for the datetime column in the index of X or y.`. Is this a known issue?

Reply:
The log is out of order. XgBoost is fine. But I set the index to None because ARIMA takes about a minute to error out on this dataset. Improving ARIMA is tracked #2611 . I'll see what I can do about the log but I think that's a separate bug!